plugin_tooling/src-lang/melnorme/lang/tooling/structure/SourceFileStructure_Default.java

```diff
@@ -38,33 +34,22 @@ public boolean equals(Object obj) {
 		
 		SourceFileStructure other = (SourceFileStructure) obj;
 		
-		return 
-			areEqual(location, other.location) &&
-			areEqual(children, other.children);
+		return
+				areEqual(children, other.children);
 	}
 	
 	@Override
 	public int hashCode() {
```

Comment:
Is `equals` and `hashCode` really necessary here? If I am not mistaken, it is not. This is rather confusing as it gives the developer the impression that this part is important and he has to care about it. My preferred way, although a bit radical, is to throw `RuntimeException`s any time you call `equals` or `hashCode` s.t. you cannot use those methods in the wrong context.